pkg/controller/subjectpermission/subjectpermission_controller.go

```diff
@@ -146,11 +149,14 @@ func (r *ReconcileSubjectPermission) Reconcile(request reconcile.Request) (recon
 		subjectName := clusterRBName[1]
 
 		// create a new clusterRoleBinding on cluster
-		newCRB := newClusterRoleBinding(clusterRoleName, subjectName)
+		newCRB := newClusterRoleBinding(clusterRoleName, subjectName, instance.Spec.SubjectKind)
 		err := r.client.Create(context.TODO(), newCRB)
 		if err != nil {
-			// calls on helper function to update the condition of the groupPermission object
-			instance := updateCondition(instance, "Unable to create ClusterRoleBinding: "+err.Error(), clusterRoleName, true, managedv1alpha1.SubjectPermissionFailed)
+			var clusterRoleNames []string
+			clusterRoleNames = append(clusterRoleNames, clusterRoleName)
+
+			// update the condition if creation of a ClusterRoleBinding has failed
+			instance := controllerutil.UpdateCondition(instance, "Unable to create ClusterRoleBinding: "+err.Error(), clusterRoleNames, true, managedv1alpha1.SubjectPermissionFailed)
```

Comment:
I'd like to see the pattern from namespace controller using `fmt.Sprintf` here as well.

Reply:
ack